framework/scripts/generate_test_list.py

```diff
@@ -75,23 +75,40 @@ def main():
     # generate the .cpp file
     with open(cpp_file, 'w') as f:
         f.write('#include <span>\n')
-        f.write(f'#include "{h_file_basename}"\n\n')
-        f.write('using namespace std;\n\n')
+        f.write(f'#include "{h_file_basename}"\n')
+        f.write('#include "sandstone_tests.h"\n')
+        f.write('#include "SelectorFactory.h"\n')
         decls = [ f'extern struct test _test_{test};' for test in sorted(all_tests.keys()) ]
         f.write('\n'.join(decls))
         f.write('\n\n')
+
         for name, tests in test_lists.items():
-            f.write(f'static constexpr struct test * const {name}_test_list[] = {{')
+            f.write(f'static constexpr struct test * const {name}_test_list_[] = {{')
             test_list = [ f'    &_test_{test}' for test in tests ]
             f.write(',\n'.join(test_list))
             f.write('\n};\n\n')
+            f.write(f'static const struct TestList {name}_test_list = {{ {"cpu_" + name if name != "default" else 0}, "{name}", {name}_test_list_ }};\n\n')
+
+        f.write('static const std::vector<const TestList*> test_lists = {\n')
```

Comment:
Same problem as before: this is a static non-POD, so it runs too early. How about a plain C array?

This is unfortunately not a theoretical or academic exercise. Looking at the generated code, I see:
```
Disassembly of section .text.startup:

0000000000000000 <_GLOBAL__sub_I_sandstone_test_lists.cpp>:
   0:   55                      push   %rbp
   1:   48 8d 05 00 00 00 00    lea    0x0(%rip),%rax        # 8 <_GLOBAL__sub_I_sandstone_test_lists.cpp+0x8>
                        4: R_X86_64_PC32        .data.rel.ro.local-0x4
   8:   48 8d 0d 00 00 00 00    lea    0x0(%rip),%rcx        # f <_GLOBAL__sub_I_sandstone_test_lists.cpp+0xf>
                        b: R_X86_64_PC32        .data.rel.ro.local+0x5c
   f:   c4 e1 f9 6e c1          vmovq  %rcx,%xmm0
```
There's an AVX instruction there.